controller/src/main/java/io/pravega/controller/server/SegmentHelper.java

```diff
@@ -173,83 +186,38 @@ public SegmentHelper(final ConnectionFactory clientCF, HostControllerStore hostS
      * @param clientRequestId     client-generated id for end-to-end tracing
      * @return void
      */
-    public CompletableFuture<Boolean> sealSegment(final String scope,
+    public CompletableFuture<Void> sealSegment(final String scope,
                                                   final String stream,
                                                   final long segmentId,
                                                   String delegationToken,
                                                   final long clientRequestId) {
         final Controller.NodeUri uri = getSegmentUri(scope, stream, segmentId);
         final String qualifiedName = getQualifiedStreamSegmentName(scope, stream, segmentId);
-        return sealSegment(qualifiedName, uri, delegationToken, clientRequestId);
-    }
-
-    private CompletableFuture<Boolean> sealSegment(final String qualifiedName,
-                                                   final Controller.NodeUri uri,
-                                                   final String delegationToken,
-                                                   long clientRequestId) {
         final WireCommandType type = WireCommandType.SEAL_SEGMENT;
         RawClient connection = new RawClient(ModelHelper.encode(uri), connectionFactory);
         final long requestId = connection.getFlow().asLong();
 
-        return connection.sendRequest(requestId, new WireCommands.SealSegment(requestId, qualifiedName, delegationToken))
-                .thenApply(r -> transformReply(r, connection, qualifiedName, WireCommands.SealSegment.class, type));
+        return sendRequest(connection, requestId, new WireCommands.SealSegment(requestId, qualifiedName, delegationToken))
+                .thenAccept(r -> handleReply(r, connection, qualifiedName, WireCommands.SealSegment.class, type));
     }
 
-    public CompletableFuture<UUID> createTransaction(final String scope,
+    public CompletableFuture<Void> createTransaction(final String scope,
                                                      final String stream,
                                                      final long segmentId,
                                                      final UUID txId,
                                                      String delegationToken) {
         final Controller.NodeUri uri = getSegmentUri(scope, stream, segmentId);
         final String transactionName = getTransactionName(scope, stream, segmentId, txId);
-
-        final CompletableFuture<UUID> result = new CompletableFuture<>();
         final WireCommandType type = WireCommandType.CREATE_SEGMENT;
-        final FailingReplyProcessor replyProcessor = new FailingReplyProcessor() {
```

Comment:
By deleting all the FailingReplyProcessor` per method, we simplify greatly this class; this is good. However, in the deleted code there were log messages that helped us to see the method executed (e.g., `createTransaction`), as well as the reason for the failure (e.g., `connectionDropped`). Are we loosing this kind of detailed info in our logs with this PR?

Reply:
The logs are present in handleReply method. 
I have added more logs for expected failing cases as well. 
Thanks